### PR TITLE
initial-implementation

### PR DESCRIPTION
**Pull request by sbuliarca**
Implemented CLI that saves all the files recursivelly in a folder through the archive-api

### CODE REVIEW

**Review by sbramin — APPROVED** [review]
Who's using this?  Could it just be bundled into the finance-fulfulment-arcive-api repo? OLGTM

**Review by MeghunMak — APPROVED** [review]

File: go.mod [hunk 1]
```diff
@@ -0,0 +1,31 @@
+module github.com/utilitywarehouse/finance-fulfilment-archive-api-cli
```
Comment (catalin-ilea) [inline go.mod:1]: Are we using do modules and pinning versions now? Do we need a team discussion around this?

Comment (sbuliarca) [reply go.mod:1]: Probably we do need a team discussion around it.
I wanted to exercise go modules first hand, that's why I used it here.

**Comment by sbuliarca** [comment]
> Who's using this? Could it just be bundled into the finance-fulfulment-arcive-api repo? OLGTM

It will be used from the exstream jobs. The idea is that we'll release the built binary as a GH release on every master merge and we'll download the binary from the exstream script. This is why it is in a new repo that is public, so that we can download it easily.

File: internal/ffaac/files_saver_worker.go [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+			} else {
+				return
+			}
+		}
+	}
+}
+
+func (f *fileSaverWorker) sendFileToArchiveAPI(ctx context.Context, fileName string) error {
+	logrus.Infof("Processing file %s", fileName)
+	file, err := os.Open(filepath.Join(f.basedir, fileName))
```
Comment (catalin-ilea) [inline internal/ffaac/files_saver_worker.go:40]: file.Close() after you're done?

File: internal/ffaac/files_saver_worker.go [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+	if err != nil {
+		return errors.Wrapf(err, "failed to open file %s", fileName)
```
Comment (catalin-ilea) [inline internal/ffaac/files_saver_worker.go]: Go 1.3 is out, shall we try out the new error wrapping?

File: go.mod (hunk 1 above)
Comment (catalin-ilea) [reply go.mod:1]: Know what you mean, but we should discuss it first before getting this out in the wild.

File: internal/ffaac/files_saver_worker.go (hunk 3 above)
Comment (sbuliarca) [reply internal/ffaac/files_saver_worker.go]: Yep, good idea

**Review by catalin-ilea — APPROVED** [review]

**Pull request merged**